src/py/aspen/covidhub_import/implementation.py

```diff
@@ -0,0 +1,136 @@
+import warnings
+from typing import Iterable, Mapping
+
+from covid_database import init_db as Cinit_db
+from covid_database import SqlAlchemyInterface as CSqlAlchemyInterface
+from covid_database import util as Cutil
+from covid_database.models.enums import ConsensusGenomeStatus
+from covid_database.models.ngs_sample_tracking import (
+    ConsensusGenome,
+    CZBID,
+    DphCZBID,
+    Project,
+    SampleFastqs,
+)
+from sqlalchemy import and_, or_
+from sqlalchemy.orm import joinedload, Session
+
+from aspen.database.connection import session_scope, SqlAlchemyInterface
+from aspen.database.models import Group, Sample, UploadedPathogenGenome
+
+
+def covidhub_interface_from_secret(secret_id: str) -> CSqlAlchemyInterface:
+    interface = Cinit_db(Cutil.get_db_uri(secret_id))
+    return interface
+
+
+def import_project(
+    interface: SqlAlchemyInterface,
+    covidhub_secret_id: str,
+    rr_project_id: str,
+):
+    covidhub_interface = covidhub_interface_from_secret(covidhub_secret_id)
+    covidhub_session: Session = covidhub_interface.make_session()
+
+    with session_scope(interface) as session:
+        project = (
+            covidhub_session.query(Project)
+            .filter(Project.rr_project_id == rr_project_id)
+            .one()
+        )
+
+        group = (
+            session.query(Group)
+            .filter(Group.name == project.originating_lab)
+            .one_or_none()
+        )
+        if group is None:
+            # copy the project info into the "group"
+            group = Group(
+                name=project.originating_lab,
+                address=project.originating_address,
+                # FIXME: this needs to be updated.
+                email="FAKE_EMAIL@gmail.com",
+            )
+            session.add(group)
+
+        consensus_genomes: Iterable[ConsensusGenome] = (
+            covidhub_session.query(ConsensusGenome)
+            .join(SampleFastqs, CZBID, Project)
+            .distinct(CZBID.id)
+            .filter(Project.rr_project_id == rr_project_id)
+            .filter(
+                or_(
+                    ConsensusGenome.status == ConsensusGenomeStatus.PASSED_AUTO,
+                    ConsensusGenome.status == ConsensusGenomeStatus.PASSED_MANUAL,
+                )
+            )
+            .order_by(
+                CZBID.id.desc(),
+                ConsensusGenome.recovered_sites.desc(),
+                ConsensusGenome.id.desc(),
+            )
+            .options(
+                joinedload(ConsensusGenome.sample_fastqs)
+                .joinedload(SampleFastqs.czb_id)
+                .joinedload(CZBID.genome_submission_info),
+            )
+        )
+
+        external_accessions = {
+            consensus_genome.sample_fastqs.czb_id.external_accession
+            for consensus_genome in consensus_genomes
+            if isinstance(consensus_genome.sample_fastqs.czb_id, DphCZBID)
```

Comment:
Maybe this is getting filtered out at a different step but it seems odd to import samples from collaborators but not include the external accession. Is the idea that we just won't specify any project ids that aren't or public DPH projects when running this script?